### PR TITLE
chore: publish official jammy image

### PR DESCRIPTION
**Pull request by rwoll**
Resolves https://github.com/microsoft/playwright-python/issues/1294.

For now, we leave `focal` as the default docker image. When v1.25 is announced, we can notify users the default base image will be changing in some future release (v1.26?) to jammy. (We should coordinate this across language ports.)

As of this writing, Python 3.10 is what ships in our jammy image.

Testing:
- Until jammy is our default base image, we'll run our tests in both focal and jammy in CI.
- This repo's tests covers amd; internal bots will cover arm.

### CODE REVIEW

**Comment by rwoll** [comment]
debugging all the webkit failures in the jammy run

**Comment by rwoll** [comment]
Looks like WK is hanging (see logs below):

```
DEBUG="pw:*" xvfb-run -- playwright screenshot --browser=wk https://example.com out.png
  pw:channel:event {
  pw:channel:event   guid: '',
  pw:channel:event   method: '__create__',
  pw:channel:event   params: {
  pw:channel:event     type: 'BrowserType',
  pw:channel:event     initializer: {
  pw:channel:event       executablePath: '/ms-playwright/chromium-1016/chrome-linux/chrome',
  pw:channel:event       name: 'chromium'
  pw:channel:event     },
  pw:channel:event     guid: 'browser-type@1cdeb79578a654e64b86774faaa87dfd'
  pw:channel:event   }
  pw:channel:event } +0ms
  pw:channel:event {
  pw:channel:event   guid: '',
  pw:channel:event   method: '__create__',
  pw:channel:event   params: {
  pw:channel:event     type: 'BrowserType',
  pw:channel:event     initializer: {
  pw:channel:event       executablePath: '/ms-playwright/firefox-1335/firefox/firefox',
  pw:channel:event       name: 'firefox'
  pw:channel:event     },
  pw:channel:event     guid: 'browser-type@ab4612a0ed7a6ea5ea0aded02bdc849c'
  pw:channel:event   }
  pw:channel:event } +6ms
  pw:channel:event {
  pw:channel:event   guid: '',
  pw:channel:event   method: '__create__',
  pw:channel:event   params: {
  pw:channel:event     type: 'BrowserType',
  pw:channel:event     initializer: {
  pw:channel:event       executablePath: '/ms-playwright/webkit-1685/pw_run.sh',
  pw:channel:event       name: 'webkit'
  pw:channel:event     },
  pw:channel:event     guid: 'browser-type@6263e047401db84e56e72cc190245274'
  pw:channel:event   }
  pw:channel:event } +0ms
  pw:channel:event {
  pw:channel:event   guid: '',
  pw:channel:event   method: '__create__',
  pw:channel:event   params: {
  pw:channel:event     type: 'Android',
  pw:channel:event     initializer: {},
  pw:channel:event     guid: 'android@f8fdd5413d10e80a9f8a7d0be8568354'
  pw:channel:event   }
  pw:channel:event } +2ms
  pw:channel:event {
  pw:channel:event   guid: '',
  pw:channel:event   method: '__create__',
  pw:channel:event   params: {
  pw:channel:event     type: 'Electron',
  pw:channel:event     initializer: {},
  pw:channel:event     guid: 'electron@8f0afa085c3aa8597e559d11896add96'
  pw:channel:event   }
  pw:channel:event } +0ms
  pw:channel:event {
  pw:channel:event   guid: '',
  pw:channel:event   method: '__create__',
  pw:channel:event   params: {
  pw:channel:event     type: 'LocalUtils',
  pw:channel:event     initializer: {},
  pw:channel:event     guid: 'localUtils@d95a741fbd1bc70d1e652aaa48d284c1'
  pw:channel:event   }
  pw:channel:event } +1ms
  pw:channel:event {
  pw:channel:event   guid: '',
  pw:channel:event   method: '__create__',
  pw:channel:event   params: {
  pw:channel:event     type: 'Selectors',
  pw:channel:event     initializer: {},
  pw:channel:event     guid: 'selectors@b131b79397db694b31b2c2c0d7748261'
  pw:channel:event   }
  pw:channel:event } +1ms
  pw:channel:event {
  pw:channel:event   guid: '',
  pw:channel:event   method: '__create__',
  pw:channel:event   params: {
  pw:channel:event     type: 'Playwright',
  pw:channel:event     initializer: {
  pw:channel:event       chromium: [Object],
  pw:channel:event       firefox: [Object],
  pw:channel:event       webkit: [Object],
  pw:channel:event       android: [Object],
  pw:channel:event       electron: [Object],
  pw:channel:event       utils: [Object],
  pw:channel:event       deviceDescriptors: [Array],
  pw:channel:event       selectors: [Object]
  pw:channel:event     },
  pw:channel:event     guid: 'Playwright'
  pw:channel:event   }
  pw:channel:event } +4ms
  pw:api =>  started +0ms
  pw:channel:command {
  pw:channel:command   id: 1,
  pw:channel:command   guid: 'browser-type@6263e047401db84e56e72cc190245274',
  pw:channel:command   method: 'launch',
  pw:channel:command   params: { ignoreAllDefaultArgs: false, headless: true }
  pw:channel:command } +0ms
  pw:browser <launching> /ms-playwright/webkit-1685/pw_run.sh --inspector-pipe --headless --no-startup-window +0ms
  pw:browser <launched> pid=2436 +25ms
  pw:protocol SEND ► {"id":1,"method":"Playwright.enable"} +0ms
  pw:browser [pid=1514] <forecefully close> +3m
  pw:browser [pid=1514] <kill> +2ms
  pw:browser [pid=1514] <will force kill> +1ms
  pw:browser [pid=1514] <kill> +2ms
  pw:browser [pid=1514] <will force kill> +0ms
  pw:browser [pid=1514] <process did exit: exitCode=null, signal=SIGKILL> +9ms
  pw:browser [pid=1514] starting temporary directories cleanup +3ms
  pw:browser [pid=1514] finished temporary directories cleanup +11ms
  pw:browser [pid=1514] <gracefully close end> +1ms
  pw:browser [pid=1975] <forecefully close> +3m
  pw:browser [pid=1975] <kill> +1ms
  pw:browser [pid=1975] <will force kill> +0ms
  pw:browser [pid=1975] <process did exit: exitCode=null, signal=SIGKILL> +2ms
  pw:browser [pid=1975] starting temporary directories cleanup +1ms
  pw:browser [pid=1975] <kill> +3ms
  pw:browser [pid=1975] <skipped force kill spawnedProcess.killed=false processClosed=true> +1ms
  pw:browser [pid=1975] finished temporary directories cleanup +5ms
  pw:browser [pid=1975] <gracefully close end> +0ms
  pw:browser [pid=2436] <gracefully close start> +3m
  pw:browser [pid=2436] <kill> +6ms
  pw:browser [pid=2436] <will force kill> +4ms
  pw:browser [pid=2436] <process did exit: exitCode=null, signal=SIGKILL> +7ms
  pw:browser [pid=2436] starting temporary directories cleanup +0ms
  pw:browser [pid=2436] finished temporary directories cleanup +7ms
  pw:browser [pid=2436] <gracefully close end> +0ms
  pw:channel:response {
  pw:channel:response   id: 1,
  pw:channel:response   error: {
  pw:channel:response     error: {
  pw:channel:response       message: 'Timeout 180000ms exceeded.\n' +
  pw:channel:response         '=========================== logs ===========================\n' +
  pw:channel:response         '<launching> /ms-playwright/webkit-1685/pw_run.sh --inspector-pipe --headless --no-startup-window\n' +
  pw:channel:response         '<launched> pid=2436\n' +
  pw:channel:response         '============================================================',
  pw:channel:response       stack: 'TimeoutError: Timeout 180000ms exceeded.\n' +
  pw:channel:response         '=========================== logs ===========================\n' +
  pw:channel:response         '<launching> /ms-playwright/webkit-1685/pw_run.sh --inspector-pipe --headless --no-startup-window\n' +
  pw:channel:response         '<launched> pid=2436\n' +
  pw:channel:response         '============================================================\n' +
  pw:channel:response         '    at ProgressController.run (/root/playwright/playwright/driver/package/lib/server/progress.js:96:26)\n' +
  pw:channel:response         '    at WebKit.launch (/root/playwright/playwright/driver/package/lib/server/browserType.js:86:38)\n' +
  pw:channel:response         '    at BrowserTypeDispatcher.launch (/root/playwright/playwright/driver/package/lib/server/dispatchers/browserTypeDispatcher.js:59:40)\n' +
  pw:channel:response         '    at DispatcherConnection.dispatch (/root/playwright/playwright/driver/package/lib/server/dispatchers/dispatcher.js:397:46)',
  pw:channel:response       name: 'TimeoutError'
  pw:channel:response     }
  pw:channel:response   }
  pw:channel:response } +0ms
  pw:api <=  failed +3m
: Timeout 180000ms exceeded.
=========================== logs ===========================
<launching> /ms-playwright/webkit-1685/pw_run.sh --inspector-pipe --headless --no-startup-window
<launched> pid=2436
============================================================
    at captureRawStack (/root/playwright/playwright/driver/package/lib/utils/stackTrace.js:60:17) {
  name: 'TimeoutError'
}
```

But the same works:

- in Docker jammy image on arm
- in non-Docker amd jammy (i.e. main repo https://github.com/microsoft/playwright/runs/7655956813?check_suite_focus=true)

**Comment by rwoll** [comment]
I tried repro'ing the above in the our official node jammy image, and it looks like it yells at a missing dep:

```
$ docker run -it --pull=always mcr.microsoft.com/playwright:next-jammy  bash
next-jammy: Pulling from playwright
Digest: sha256:d02d77f3a3c4dbe9ed1f5f1cc38835afc6418ad7b534885ac0ba855aa4c10a07
Status: Image is up to date for mcr.microsoft.com/playwright:next-jammy
root@388cc365118e:/# mkdir foo
root@388cc365118e:/# cd foo/
root@388cc365118e:/foo# npm init -y
Wrote to /foo/package.json:

{
  "name": "foo",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}


root@388cc365118e:/foo# npm i playwright@next

added 2 packages, and audited 3 packages in 905ms

found 0 vulnerabilities
root@388cc365118e:/foo# npx playwright screenshot --browser=wk https://example.com example.png
object.<anonymous>: 
╔══════════════════════════════════════════════════════╗
║ Host system is missing dependencies to run browsers. ║
║ Missing libraries:                                   ║
║     libatomic.so.1                                   ║
╚══════════════════════════════════════════════════════╝
    at Object.<anonymous> (/foo/node_modules/playwright-core/cli.js:17:18)
    at Object.<anonymous> (/foo/node_modules/playwright/cli.js:17:18) {
  name: 'Error'
}
root@388cc365118e:/foo# npx playwright install --with-deps
Installing dependencies...
Get:1 https://deb.nodesource.com/node_16.x jammy InRelease [4583 B]
Get:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]                                    
Get:3 https://deb.nodesource.com/node_16.x jammy/main amd64 Packages [774 B]           
Get:4 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]                        
Get:5 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [303 kB]
Get:6 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [276 kB]
Get:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [114 kB]
Get:8 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [99.8 kB]
Get:9 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [4644 B]
Get:10 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [119 kB]
Get:11 http://archive.ubuntu.com/ubuntu jammy/multiverse amd64 Packages [266 kB]
Get:12 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages [1792 kB]
Get:13 http://archive.ubuntu.com/ubuntu jammy/restricted amd64 Packages [164 kB]            
Get:14 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages [17.5 MB]
Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [7791 B]                                                                                                      
Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [319 kB]                                                                                                      
Get:17 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [243 kB]                                                                                                        
Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [602 kB]                                                                                                            
Get:19 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [5814 B]                                                                                                      
Fetched 22.2 MB in 9s (2483 kB/s)                                                                                                                                                             
Reading package lists... Done
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
fonts-freefont-ttf is already the newest version (20120503-10build1).
fonts-liberation is already the newest version (1:1.07.4-11).
fonts-noto-color-emoji is already the newest version (2.034-1).
gstreamer1.0-plugins-base is already the newest version (1.20.1-1).
libasound2 is already the newest version (1.2.6.1-1ubuntu1).
libatk-bridge2.0-0 is already the newest version (2.38.0-3).
libatk1.0-0 is already the newest version (2.36.0-3build1).
libatspi2.0-0 is already the newest version (2.44.0-3).
libcairo-gobject2 is already the newest version (1.16.0-5ubuntu2).
libcairo2 is already the newest version (1.16.0-5ubuntu2).
libdbus-1-3 is already the newest version (1.12.20-2ubuntu4).
libdbus-glib-1-2 is already the newest version (0.112-2build1).
libdrm2 is already the newest version (2.4.110-1ubuntu1).
libegl1 is already the newest version (1.4.0-1).
libenchant-2-2 is already the newest version (2.3.2-1ubuntu2).
libepoxy0 is already the newest version (1.5.10-1).
libevdev2 is already the newest version (1.12.1+dfsg-1).
libfontconfig1 is already the newest version (2.13.1-4.2ubuntu5).
libgdk-pixbuf-2.0-0 is already the newest version (2.42.8+dfsg-1).
libgles2 is already the newest version (1.4.0-1).
libglib2.0-0 is already the newest version (2.72.1-1).
libglx0 is already the newest version (1.4.0-1).
libgstreamer-gl1.0-0 is already the newest version (1.20.1-1).
libgstreamer-plugins-base1.0-0 is already the newest version (1.20.1-1).
libgudev-1.0-0 is already the newest version (1:237-2build1).
libhyphen0 is already the newest version (2.8.8-7build2).
libicu70 is already the newest version (70.1-2).
libjpeg-turbo8 is already the newest version (2.1.2-0ubuntu1).
liblcms2-2 is already the newest version (2.12~rc1-2build2).
libmanette-0.2-0 is already the newest version (0.2.6-3build1).
libnspr4 is already the newest version (2:4.32-3build1).
libopengl0 is already the newest version (1.4.0-1).
libopenjp2-7 is already the newest version (2.4.0-6).
libopus0 is already the newest version (1.3.1-0.1build2).
libpango-1.0-0 is already the newest version (1.50.6+ds-2).
libpangocairo-1.0-0 is already the newest version (1.50.6+ds-2).
libpng16-16 is already the newest version (1.6.37-3build5).
libproxy1v5 is already the newest version (0.4.17-2).
libsecret-1-0 is already the newest version (0.20.5-2).
libsoup2.4-1 is already the newest version (2.74.2-3).
libwayland-client0 is already the newest version (1.20.0-1).
libwayland-egl1 is already the newest version (1.20.0-1).
libwayland-server0 is already the newest version (1.20.0-1).
libwebpdemux2 is already the newest version (1.2.2-2).
libwoff1 is already the newest version (1.0.2-1build4).
libx11-6 is already the newest version (2:1.7.5-1).
libx11-xcb1 is already the newest version (2:1.7.5-1).
libxcb-shm0 is already the newest version (1.14-3ubuntu3).
libxcb1 is already the newest version (1.14-3ubuntu3).
libxcomposite1 is already the newest version (1:0.4.5-1build2).
libxcursor1 is already the newest version (1:1.2.0-2build4).
libxdamage1 is already the newest version (1:1.1.5-2build2).
libxext6 is already the newest version (2:1.3.4-1build1).
libxfixes3 is already the newest version (1:6.0.0-1).
libxi6 is already the newest version (2:1.8-1build1).
libxkbcommon0 is already the newest version (1.4.0-1).
libxrandr2 is already the newest version (2:1.5.2-1build1).
libxrender1 is already the newest version (1:0.9.10-1build4).
libxslt1.1 is already the newest version (1.1.34-4build2).
libxtst6 is already the newest version (2:1.2.3-1build4).
xfonts-scalable is already the newest version (1:1.0.3-1.2ubuntu1).
fonts-ipafont-gothic is already the newest version (00303-21ubuntu1).
fonts-tlwg-loma-otf is already the newest version (1:0.7.3-1).
fonts-unifont is already the newest version (1:14.0.01-1).
fonts-wqy-zenhei is already the newest version (0.9.45-8).
libffi7 is already the newest version (3.3-5ubuntu1).
libx264-163 is already the newest version (2:0.163.3060+git5db6aa6-2build1).
xfonts-cyrillic is already the newest version (1:1.0.5).
gstreamer1.0-plugins-good is already the newest version (1.20.3-0ubuntu1).
libcups2 is already the newest version (2.4.1op1-1ubuntu4.1).
libfreetype6 is already the newest version (2.11.1+dfsg-1ubuntu0.1).
libgbm1 is already the newest version (22.0.5-0ubuntu0.1).
libgstreamer1.0-0 is already the newest version (1.20.3-0ubuntu1).
libgtk-3-0 is already the newest version (3.24.33-1ubuntu2).
libharfbuzz-icu0 is already the newest version (2.7.4-1ubuntu3.1).
libharfbuzz0b is already the newest version (2.7.4-1ubuntu3.1).
libjavascriptcoregtk-4.0-18 is already the newest version (2.36.4-0ubuntu0.22.04.1).
libnotify4 is already the newest version (0.7.9-3ubuntu5.22.04.1).
libnss3 is already the newest version (2:3.68.2-0ubuntu1.1).
libxml2 is already the newest version (2.9.13+dfsg-1ubuntu0.1).
ffmpeg is already the newest version (7:4.4.2-0ubuntu0.22.04.1).
gstreamer1.0-libav is already the newest version (1.20.3-0ubuntu1).
gstreamer1.0-plugins-bad is already the newest version (1.20.3-0ubuntu1).
xvfb is already the newest version (2:21.1.3-2ubuntu2.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@388cc365118e:/foo# npx playwright screenshot --browser=wk https://example.com example.png
object.<anonymous>: 
╔══════════════════════════════════════════════════════╗
║ Host system is missing dependencies to run browsers. ║
║ Missing libraries:                                   ║
║     libatomic.so.1                                   ║
╚══════════════════════════════════════════════════════╝
    at Object.<anonymous> (/foo/node_modules/playwright-core/cli.js:17:18)
    at Object.<anonymous> (/foo/node_modules/playwright/cli.js:17:18) {
  name: 'Error'
}
```

cc: @mxschmitt

**Comment by rwoll** [comment]
re: https://github.com/microsoft/playwright-python/pull/1477#issuecomment-1204467998

even if I install `apt install libatomic1` and the missing dep error goes away, webkit hangs, so this does not look python-container specific.

**Comment by rwoll** [comment]
hid some debug comments; jammy wk failures will likely be resolved by https://github.com/microsoft/playwright/issues/16228

File: utils/docker/publish_docker.sh [hunk 1]
```diff
@@ -92,8 +101,10 @@ publish_docker_manifest () {
     TAGS=("${BIONIC_TAGS[@]}")
   elif [[ "$FLAVOR" == "focal" ]]; then
     TAGS=("${FOCAL_TAGS[@]}")
+  elif [[ "$FLAVOR" == "jammy" ]]; then
+    TAGS=("${JAMMY_TAGS[@]}")
   else
-    echo "ERROR: unknown flavor - $FLAVOR. Must be either 'bionic' or 'focal'"
+    echo "ERROR: unknown flavor - $FLAVOR. Must be either 'bionic', 'focal', or 'jammy'"
```
Comment (mxschmitt) [inline utils/docker/publish_docker.sh:107]: you miss in the end the actual publishing part.

Comment (rwoll) [reply utils/docker/publish_docker.sh:107]: 😄 fixed, I think! Would've found out soon enough haha

File: utils/docker/publish_docker.sh [hunk 2]
```diff
@@ -117,4 +128,6 @@ publish_docker_manifest bionic amd64
 
 publish_docker_images_with_arch_suffix focal amd64
 publish_docker_images_with_arch_suffix focal arm64
+publish_docker_images_with_arch_suffix jammy amd64
```
Comment (mxschmitt) [inline utils/docker/publish_docker.sh:131]: publish_docker_manifest jammy amd64 arm64

Comment (mxschmitt) [reply utils/docker/publish_docker.sh:131]: Thats missing

**Review by mxschmitt — APPROVED** [review]

**Review by mxschmitt — APPROVED** [review]

**Comment by retrry** [comment]
Are jammy images built and named correctly? From changelog I see that `mcr.microsoft.com/playwright/python:v1.25.0-jammy` image should be available, but I can't see it here: https://mcr.microsoft.com/en-us/product/playwright/python/tags with that tag.

**Comment by aslushnikov** [comment]
> Are jammy images built and named correctly?

Should be! Please file an issue showing what you do and how it fails!

**Comment by retrry** [comment]
@aslushnikov created an issue: https://github.com/microsoft/playwright-python/issues/1518

**Pull request merged**